### ISSUE TITLE
False positive with goto: warning CS0162: Unreachable code detected

### ISSUE THREAD

**Issue by ilyfairy** [dotnet/roslyn#68437]
**Version Used**: Version 17.7.0 Preview 1.0

**Steps to Reproduce**:

[sharplab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQBED2BjOBbCAOxgHk4YAbAS0IgBoYQIAPABwqhroBMQBqAD4ABAEwBGALAAoIQGYABKPkBheQG9p8rYoVCALPICyACgCU6zdqsBzTDEzyi3ANzSAkG5ox5UV1LdOIH4eUPIAvPIiAAzBbkJiAJzGUKZ+VgC+0ulAA===)
```cs
using System;
public class C {
    public void M() {
        goto end;
        int a;
    end:;
	a = 20;
	Console.WriteLine(a);
    }
}
```

**Expected Behavior**:

No warning.

**Actual Behavior**:

`warning CS0162: Unreachable code detected`
